src/test/scala/com/agar/core/region/RegionTest.scala

```diff
@@ -23,25 +27,243 @@ class RegionTest(_system: ActorSystem)
+
+    val player = journal.expectMsgPF(500 millis) { case WorldState(newPlayers, _) => newPlayers.toList.head._1 }
+
+    region ! Move(player, Vector2d(10, 6), 0)
+
+    bridge.expectMsgPF(500 millis) {
+      case Virtual(RegisterPlayer(p, PlayerState(Vector2d(10, 4), 0, Vector2d(0, 0), false))) => player === p
+    }
+
+    val expectedCorrectMessage = bridge.expectMsgPF(500 millis) {
+      case Virtual(Destroy(p)) => player === p
+    }
+
+    expectedCorrectMessage should be(true)
+  }
+
+  "kill a player in the frontier" in {
+    implicit val context: AgarSystem = () => 2 seconds
+
+    val journal = TestProbe()
+    val bridge = TestProbe()
+    val region = system.actorOf(Region.props(10, 10, 5)(journal.ref, bridge.ref))
+
+    region ! InitRegion(0, 0)
+
+    journal.expectMsgPF(500 millis) {
+      case Initialized(_, _) => ()
+    }
+
+    region ! CreatePlayer(PlayerState(Vector2d(10, 4), 0, Vector2d(0, 0)))
+    region ! GetEntitiesAOISet
+
+    val player = journal.expectMsgPF(500 millis) { case WorldState(newPlayers, _) => newPlayers.toList.head._1 }
+
+    region ! Killed(player)
+
+    bridge.expectMsgPF(500 millis) {
+      case Virtual(RegisterPlayer(p, PlayerState(Vector2d(10, 4), 0, Vector2d(0, 0), false))) => player === p
+    }
+
+    val expectedCorrectMessage = bridge.expectMsgPF(500 millis) {
+      case Virtual(Killed(p)) => player === p
+    }
+
+    expectedCorrectMessage should be(true)
+  }
+
+  "move a player out the frontier and the region" in {
+    implicit val context: AgarSystem = () => 2 seconds
+
+    val journal = TestProbe()
+    val bridge = TestProbe()
+    val region = system.actorOf(Region.props(10, 10, 5)(journal.ref, bridge.ref))
+
+    region ! InitRegion(0, 0)
+
+    journal.expectMsgPF(500 millis) {
+      case Initialized(_, _) => ()
+    }
+
+    region ! CreatePlayer(PlayerState(Vector2d(10, 7), 0, Vector2d(0, 0)))
+    region ! GetEntitiesAOISet
+
+    val player = journal.expectMsgPF() { case WorldState(newPlayers, _) => newPlayers.toList.head._1 }
+
+    region ! Move(player, Vector2d(10, -7), 0)
+
+    val expectedCorrectMessage = bridge.expectMsgPF(500 millis) {
+      case Virtual(CreatePlayer(PlayerState(Vector2d(10, -7), 0, Vector2d(0, 0), false))) => true
+    }
+
+    expectedCorrectMessage should be(true)
+  }
+
+  "initialize a fresh region with one fresh energy in the frontier" in {
+    implicit val context: AgarSystem = () => 2 seconds
+
+    val journal = TestProbe()
+    val bridge = TestProbe()
+    val region = system.actorOf(Region.props(10, 10, 10)(journal.ref, bridge.ref))
+
+    region ! InitRegion(0, 1)
+
+    val expectedCorrectInit = journal.expectMsgPF() {
+      case Initialized(players, energies) => {
+        players.size === 0 && energies.size === 1
+      }
+    }
+
+    expectedCorrectInit should be(true)
+
+    bridge.expectMsgPF(500 millis) {
+      case Virtual(RegisterEnergy(_, _)) => true
+    }
+
+  }
+
+  "consume an energy in a frontier" in {
```

Comment:
I think we are missing a test: kill a player in the region.
This is not an important test but we have the "kill a player in the frontier" so for me we should have the "kill a player in the region" too.

Reply:
Tests added

Reply:
thx !